src/settleAll.ts

```diff
@@ -17,14 +16,24 @@ export interface SettledPromises<T, V> {
 export async function settleAll<T, V>(
   promises: readonly Promise<T>[],
   // tslint:disable-next-line:no-any (no way to guarantee error typings)
-  errFn: (err: any) => V = err => err,
+  errFn?: (err: any) => V,
+): Promise<SettledPromises<T, V>>;
+export async function settleAll<T, V>(
+  promises: readonly Promise<T>[],
+  // tslint:disable-next-line:no-any (no way to guarantee error typings)
+  errFn?: (err: any, ind: number) => V,
```

Comment:
you have to reverse the order of these overloads or it will always use the first one iirc